service/controller/clusterapi/v28/machine_deployment_resource_set.go

```diff
@@ -83,9 +87,26 @@ func NewMachineDeploymentResourceSet(config MachineDeploymentResourceSetConfig)
 	}
 
 	initCtxFunc := func(ctx context.Context, obj interface{}) (context.Context, error) {
+		var cr v1alpha1.Cluster
```

Comment:
I think this should be done in a resource and we should maybe drop the `initCtxFunc` thingy completely. I am not sure that thing was really useful at all.

Reply:
I don't like passing information "up" the context in general, but I think we should address it other way so nothing against that if technically possible.

Reply:
What do you mean with passing up?